eclipse-theia/theia
# Simplify localization for vscode related strings

#### What it does

Related to the discussions that occurred in #10106 and #10299, 

Heavily simplifies the localization process for strings that are contained inside of vscode. Additionally removes the coupling between vscode internals and Theia's translations calls.

- [x] [CQ for `nls.metadata.json`](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23798)

#### How it works

Utilizes the `nls.metadata.json` that is generated by the vscode build process (in this case the 1.53.2 version of vscode). It can be produced by running `yarn && yarn gulp vscode-linux-x64` inside of the vscode repository. This files contains every default value and its key of the vscode `nls.localize` calls that occur inside of the vscode codebase. With a little bit of transformation work, we can produce a map that maps every default value to the associated key.

This has the drawback of commiting/polluting the Theia repository with this generated file. I'm open to suggestions on how to improve this.

#### How to test

1. Open Theia in english and observe that everything continues running as usual
2. Install a language pack of your choice and run the `Configure Display Language` command.
3. Assert that the application is (mostly) translated into the target language, even without explicitly referencing vscode keys.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
